awx/ui/client/src/scheduler/factories/schedule-post.factory.js

```diff
@@ -64,43 +65,65 @@ export default
                     }
                     if(_.has(promptData, 'prompts.inventory.value') && _.get(promptData, 'launchConf.ask_inventory_on_launch')){
-                        schedule.inventory = promptData.prompts.inventory.templateDefault.id === promptData.prompts.inventory.value.id ? null : promptData.prompts.inventory.value.id;
+                        scheduleData.inventory = promptData.launchConf.defaults.inventory && promptData.launchConf.defaults.inventory.id === promptData.prompts.inventory.value.id ? null : promptData.prompts.inventory.value.id;
                     }
                     if(_.has(promptData, 'prompts.diffMode.value') && _.get(promptData, 'launchConf.ask_diff_mode_on_launch')){
-                        schedule.diff_mode = promptData.prompts.diffMode.templateDefault === promptData.prompts.diffMode.value ? null : promptData.prompts.diffMode.value;
+                        scheduleData.diff_mode = promptData.launchConf.defaults.diff_mode && promptData.launchConf.defaults.diff_mode === promptData.prompts.diffMode.value ? null : promptData.prompts.diffMode.value;
                     }
-                    // Credentials gets POST'd to a separate endpoint
-                    // if($scope.promptData.launchConf.ask_credential_on_launch){
-                    //     jobLaunchData.credentials = [];
-                    //     promptData.credentials.value.forEach((credential) => {
-                    //         jobLaunchData.credentials.push(credential.id);
-                    //     });
-                    // }
                 }
 
                 Rest.setUrl(url);
                 if (mode === 'add') {
-                    Rest.post(schedule)
-                        .then(() => {
-                            Wait('stop');
-                            deferred.resolve();
+                    Rest.post(scheduleData)
+                        .then(({data}) => {
+                            if(_.get(promptData, 'launchConf.ask_credential_on_launch')){
+                                // This finds the credentials that were selected in the prompt but don't occur
+                                // in the template defaults
+                                let credentialsToPost = promptData.prompts.credentials.value.filter(function(credFromPrompt) {
+                                    let defaultCreds = promptData.launchConf.defaults.credentials ? promptData.launchConf.defaults.credentials : [];
+                                    return !defaultCreds.some(function(defaultCred) {
+                                        return credFromPrompt.id === defaultCred.id;
+                                    });
+                                });
+
+                                let promises = [];
+                                let schedule = new Schedule();
```

Comment:
Is schedule reassigned at some point? If not, we could reconsider using `let`. This also applies to any identifier that is not reassigned in the pr.

Reply:
Ah good catch - i'll run back through and take a look at where we might be able to do that.